app/components/UI/LndConnectionStringInput.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+import { intlShape, injectIntl } from 'react-intl'
+import get from 'lodash.get'
+import decode from 'lndconnect/decode'
+import TextArea from 'components/UI/TextArea'
+import messages from './messages'
+
+/**
+ * @render react
+ * @name LndConnectionStringInput
+ * @example
+ * <LndConnectionStringInput
+     field="testnet"
+     id="testnet"
+     validateOnChange
+     validateOnBlur />
+ */
+class LndConnectionStringInput extends React.Component {
+  static displayName = 'LndConnectionStringInput'
+
+  static propTypes = {
+    intl: intlShape.isRequired
+  }
+
+  prettyPrint = json => {
+    try {
+      return JSON.stringify(JSON.parse(json), undefined, 4)
+    } catch (e) {
+      return json
+    }
+  }
+
+  /**
+   * Check for a valid lndconnect uri or BtcPayServer connection string.
+   * @param  {String}  value String to validate.
+   * @return {Boolean}       Boolean indicating wether the string is a valid or not.
+   */
+  validate = value => {
+    const { intl } = this.props
+    const isValid = this.isValidLndConnectUri(value) || this.isValidBtcPayConfig(value)
+    if (!isValid) {
+      return intl.formatMessage({ ...messages.invalid_lnd_connection_string })
+    }
+  }
+
+  /**
+   * Check for a valid lndconnect uri.
+   * @param  {String}  value String to validate.
+   * @return {Boolean}       Boolean indicating wether the string is a valid or not.
+   */
+  isValidLndConnectUri = value => {
+    let config = {}
+    try {
+      config = decode(value)
+      const { host } = config
+      if (host) {
+        return true
+      }
+      return false
+    } catch (e) {
+      return false
+    }
+  }
+
+  /**
+   * Check for a valid BtcPayServer connection string.
+   * @param  {String}  value String to validate.
+   * @return {Boolean}       Boolean indicating wether the string is a valid or not.
+   */
+  isValidBtcPayConfig = value => {
+    let config = {}
```

Comment:
`config` can be moved as a const inside `try`

Reply:
Updated.